src/primitives/transaction.h

```diff
@@ -527,47 +520,40 @@ class CTxForwardTransferOut : public CTxCrosschainOut
 
 class CTxScCreationOut : public CTxCrosschainOut
 {
+friend class CTransaction;
```

Comment:
It look like you need this friend declaration because you need to call `GenerateScId` when you call `UpdateHash `on `CTransaction`.
Why instead of storing the `generatedScId` into the `CTxScCreationOut ` don't we store it into a vector into CTransaction? After all `GetScId()` looks called only from the corresponding method in CTransation

Reply:
As agree AKF, let's postpone it to the final review to tackle it along with other CTx/CScCert related changes